sql/compression_defaults.sql

```diff
@@ -0,0 +1,291 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+CREATE OR REPLACE FUNCTION _timescaledb_functions.get_segmentby_defaults(
+    relation regclass
+)
+    RETURNS JSONB LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    _table_name NAME;
+    _schema_name NAME;
+    _hypertable_row _timescaledb_catalog.hypertable;
+    _segmentby NAME;
+    _cnt int;
+BEGIN
+    SELECT n.nspname, c.relname INTO STRICT _schema_name, _table_name
+    FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.oid = c.relnamespace)
+    WHERE c.oid = relation;
+
+    SELECT * INTO STRICT _hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.table_name = _table_name AND h.schema_name = schema_name;
+
+    --STEP 1 if column stats exist use unique indexes
```

Comment:
This picks one column from the unique index only if that column has enough rows-per-value according to statistics so that a compressed row has many uncompressed rows. This only work for multi-column indexes where the column that is picked is /not/ unique. 

If there are multiple unique indexes we pick the column that comes first in the index (has the minimum position). I added a comment